src/go/MONIT/alert.go

```diff
@@ -0,0 +1,556 @@
+func sortAlert() {
+
+	switch strings.ToLower(sortLabel) {
+	case "severity":
+		sort.Sort(severitySorter(allAlertData))
+	case "starts":
+		sort.Sort(startAtSorter(allAlertData))
+	case "ends":
+		sort.Sort(endsAtSorter(allAlertData))
+	case "duration":
+		sort.Sort(durationSorter(allAlertData))
+	default:
+		return
+	}
+}
+
+//Function for printing alerts in JSON format
+func jsonPrint() {
+
+	var filteredData []alertDataJSON
+	var temp alertDataJSON
+
+	for _, each := range allAlertData {
+		if filteredAlerts[each.Name] == 1 {
+			temp.Name = each.Name
+			temp.Service = each.Service
+			temp.Severity = each.Service
+			temp.Tag = each.Tag
+			temp.Starts = timeDiff(time.Now(), each.StartsAt, 0)
+			if each.EndsAt == time.Unix(maxtstmp, 0).UTC() {
+				temp.Ends = "Undefined"
+				temp.Duration = "Undefined"
+			} else {
+				temp.Ends = timeDiff(time.Now(), each.EndsAt, 0)
+				temp.Duration = timeDiff(each.StartsAt, each.EndsAt, 1)
+			}
+
+			filteredData = append(filteredData, temp)
+		}
+	}
+
+	b, err := json.Marshal(filteredData)
+
+	if err != nil {
+		log.Printf("Unable to convert Filtered JSON Data, error: %v\n", err)
+		return
+	}
+
+	fmt.Println(string(b))
+}
+
+//Function for printing alerts in Plain text format
+func tabulate() {
+
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 8, 8, 0, '\t', 0)
+	defer w.Flush()
+
+	fmt.Fprintf(w, "\n %s\t\t%s\t\t%s\t\t%s\t\t%s\t\t%s\t\t%s\n", "NAME", "SERVICE", "TAG", "SEVERITY", "STARTS", "ENDS", "DURATION")
+
+	for _, each := range allAlertData {
+		if filteredAlerts[each.Name] == 1 {
+			fmt.Fprintf(w, " %s\t\t%s\t\t%s\t\t%s\t\t%s",
+				each.Name,
+				each.Service,
+				each.Tag,
+				each.Severity,
+				timeDiff(time.Now(), each.StartsAt, 0),
+			)
+			if each.EndsAt == time.Unix(maxtstmp, 0).UTC() {
+				fmt.Fprintf(w, "\t\t%s", "Undefined")
+				fmt.Fprintf(w, "\t\t%s\n", "Undefined")
+			} else {
+				fmt.Fprintf(w, "\t\t%s", timeDiff(time.Now(), each.EndsAt, 0))
+				fmt.Fprintf(w, "\t\t%s\n", timeDiff(each.StartsAt, each.EndsAt, 1))
+			}
+
+		}
+	}
+}
+
+//Function for printing alert's details in JSON format
+func jsonPrintDetails() {
+
+	b, err := json.Marshal(alertDetails[name])
+
+	if err != nil {
+		log.Printf("Unable to convert Detailed JSON Data, error: %v\n", err)
+		return
+	}
+
+	fmt.Println(string(b))
+}
+
+//Function to get all keys of type map[string]interface{}
+func getkeys(m map[string]interface{}) []string {
+
+	keys := make([]string, 0, len(m))
+	for key := range m {
+		keys = append(keys, key)
+	}
+	return keys
+}
+
+//Function for printing alert's details in Plain text format
+func detailPrint() {
+
+	if alert, ok := alertDetails[name]; ok {
+		labelsKeys := getkeys(alertDetails[name].Labels)
+		sort.Strings(labelsKeys)
+
+		for _, each := range labelsKeys {
+			switch each {
+			case "alertname":
+				fmt.Printf("NAME: %s\n", alertDetails[name].Labels[each])
+				fmt.Printf("LABELS\n")
+			case "service", "tag", "severity":
```

Comment:
Can these names be defined in configuration? I want to avoid hard-coding the values. If user does not provide them in config you can assign them as defaults use them here.

Reply:
Added this functionality.

Reply:
I don't see that anything is changed here, I though you'll do something like:
```
case configJSON.attributes
```
where `configJSON.attributes` can either be provided in config or set to to be a list of `"service", "tag", "severity"`.

Reply:
@vkuznet I did add configJSON.Names which is []string having ["NAMES", "LABELS", "ANNOTATIONS"]. and it uses :- 
```
for _, each := range labelsKeys {
   switch each {
      case "alertname":
	fmt.Printf("%s: %s\n", configJSON.Names[0], alertDetails[name].Labels[each])
	fmt.Printf("%s\n", configJSON.Names[1])
      case "service", "tag", "severity":
	fmt.Printf("\t%s: %s\n", each, alertDetails[name].Labels[each])
    }
}
```
Now, I have also added "service", "tag", "severity" as configJSON.attributes.